### PR TITLE
Display FPS

### PR DESCRIPTION
**Pull request by milkdropper**
Display FPS

### CODE REVIEW

File: src/libprojectM/Renderer/Renderer.cpp [hunk 1]
```diff
@@ -229,6 +236,16 @@ void Renderer::ResetTextures()
 void Renderer::SetupPass1(const Pipeline& pipeline, const PipelineContext& pipelineContext)
 {
 	totalframes++;
+	this->currentTime = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
```
Comment (revmischa) [inline src/libprojectM/Renderer/Renderer.cpp]: perhaps we could avoid doing this work if we're not outputting FPS?

File: src/libprojectM/Renderer/Renderer.cpp [hunk 2]
```diff
@@ -229,6 +236,16 @@ void Renderer::ResetTextures()
...
+	milliseconds ms = std::chrono::duration_cast<std::chrono::milliseconds>(this->currentTime - this->lastTime);
+	double diff = ms.count();
+	if (diff >= 250)
+	{
```
Comment (revmischa) [inline src/libprojectM/Renderer/Renderer.cpp]: comment explaining this block would be nice

Comment (milkdropper) [reply src/libprojectM/Renderer/Renderer.cpp]: There is already a total frame count (totalframes). The idea is to track from the initial run until we reach 250 milliseconds (1/4 of a second). To estimate the FPS I multiply the totalframes rendered in 1/4 of a second by 4. That way we get an update every 1/4 of a second.

Comment (milkdropper) [reply src/libprojectM/Renderer/Renderer.cpp]: I will put in a comment block to explain the logic.

File: src/libprojectM/Renderer/Renderer.cpp (hunk 1 above)
Comment (milkdropper) [reply src/libprojectM/Renderer/Renderer.cpp]: You are right. If FPS is not being displayed there is no need to check this timer. I will make that change.

**Comment by revmischa** [comment]
Also, isn't there a FPS counter in the SDL app? Wonder if there's a way to clean that up and have it use this code

**Comment by milkdropper** [comment]
> Also, isn't there a FPS counter in the SDL app? Wonder if there's a way to clean that up and have it use this code

Can you point me in the direction of this? I looked into projectM-sdl and only found a placeholder to display FPS. The lack of something working is why I submitted this.

Here is what projectM-sdl\Release\projectM.exe looks like after pressing F5 now.
image

**Comment by revmischa** [comment]
There sort of is in SDL but it's really more for timing the frame delays - https://github.com/projectM-visualizer/projectm/blob/master/src/projectM-sdl/projectM_SDL_main.cpp#L454

**Pull request merged**